server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1618,6 +1618,12 @@ public static BiConsumer<String, MappingParserContext> notFromDynamicTemplates(S
         };
     }
 
+    private static final IndexVersion MINIMUM_LEGACY_COMPATIBILITY_VERSION = IndexVersion.fromId(5000099);
+
+    public static TypeParser createTypeParserWithLegacySupport(BiFunction<String, MappingParserContext, Builder> builderFunction) {
+        return new TypeParser(builderFunction, MINIMUM_LEGACY_COMPATIBILITY_VERSION);
+    }
```

Comment:
This makes the code a little easier to follow: there is one lower bound legacy compatibility constant, rather than one per mapper. After all the lower bound is always v5. It's also easier to follow what mappers provide legacy support, as they must now all call this method.